### PR TITLE
  devDeps: babel-*@6 -> @babel/*@^7

### PR DESCRIPTION
**Pull request by legobeat**
- Bump and unpin eslint from 2 to 5
  - Linting fixes and config update 
- Update from webpack 2 to webpack 3
- Update from babel 6 to babel 7
  - **BREAKING**: Add peerDependency `@babel/runtime@^7.0.0` 

---

#### Blocking
- #15

### CODE REVIEW

**Comment by legobeat** [comment]
@SocketSecurity ignore rxjs@6.6.7

network access ok

**Comment by legobeat** [comment]
@SocketSecurity ignore update-browserslist-db@1.0.13

shall access ok

**Comment by legobeat** [comment]
@SocketSecurity ignore @nicolo-ribaudo/eslint-scope-5-internals@5.1.1-v1
    @SocketSecurity ignore eslint-module-utils@2.8.0
    @SocketSecurity ignore util@0.10.4
    @SocketSecurity ignore os-browserify@0.3.0
    @SocketSecurity ignore stream-browserify@2.0.2
    @SocketSecurity ignore console-browserify@1.2.0
    @SocketSecurity ignore is-async-function@2.0.0
    @SocketSecurity ignore convert-source-map@2.0.0
    @SocketSecurity ignore unicode-canonical-property-names-ecmascript@2.0.0
    @SocketSecurity ignore unicode-match-property-ecmascript@2.0.0
    @SocketSecurity ignore ms@2.1.2
    @SocketSecurity ignore browserify-zlib@0.2.0

new author, no readme ok

**Comment by legobeat** [comment]
@SocketSecurity ignore yargs-parser@7.0.0
    @SocketSecurity ignore mem@1.1.0

known issues via babel-loader, webpack

File: package.json [hunk 1]
```diff
@@ -191,6 +196,9 @@
   "license": "MIT",
   "main": "lib/index.js",
   "name": "@metamask/ethjs-contract",
+  "peerDependencies": {
+    "@babel/runtime": "^7.0.0"
```
Comment (mcmire) [inline package.json:200]: Why do we need this?

Comment (legobeat) [reply package.json:200]: See https://babeljs.io/docs/v7-migration

(`babel-runtime` v6 is already an implicit peerDep prior to this PR)

Comment (mcmire) [reply package.json:200]: Sorry, where are you looking? I see mention of plugins having a peer dependency on `@babel/core` here: https://babeljs.io/docs/v7-migration#versioningdependencies. But I don't see anything around a need to add Babel as a peer dependency if you're using Babel.

More importantly, we're just using Babel to compile this package — we're just using it as a dev dependency. Why should consumers of this package need to have `@babel/runtime` installed just to use it?

**Comment by legobeat** [comment]
@SocketSecurity ignore setimmediate@1.0.5
    @SocketSecurity ignore bs58@4.0.1
    @SocketSecurity ignore ripemd160@2.0.2
    @SocketSecurity ignore md5.js@1.3.5

new author, no readme ok

File: package.json (hunk 1 above)
Comment (legobeat) [reply package.json:200]: It's alluded to here specifically: https://babeljs.io/docs/v7-migration#babelruntime-babelplugin-transform-runtime

https://stackoverflow.com/questions/63767991/do-babel-runtime-babel-plugin-transform-runtime-in-cli-need-to-be-dependenci

This package uses `babel-plugin-transform-runtime`. Might be possible to drop after compatibility with older runtimes has been removed.

**Comment by legobeat** [comment]
@SocketSecurity ignore component-emitter@1.3.1

new author ok

**Comment by legobeat** [comment]
@SocketSecurity ignore electron-to-chromium@1.4.590

"unpublished" (bot has stale cache?)

**Comment by legobeat** [comment]
@SocketSecurity ignore ansi-regex@4.1.0

legacy via devDep `ganache-cli`

**Comment by legobeat** [comment]
@SocketSecurity ignore electron-to-chromium@1.4.591


"unpublished"

**Comment by legobeat** [comment]
@SocketSecurity ignore glob-parent@3.1.0

webpack3 legacy

**Review by kumavis — APPROVED** [review]

**Pull request merged**